File: src/main/java/com/dunamismax/site/web/AboutController.java
package com.dunamismax.site.web;

import com.dunamismax.site.config.SiteProperties;
import com.dunamismax.site.content.ContentService;
import com.dunamismax.site.content.PageMeta;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class AboutController {

  private final ContentService content;
  private final SiteProperties site;
  private final PageModel pageModel;

  public AboutController(ContentService content, SiteProperties site, PageModel pageModel) {
    this.content = content;
    this.site = site;
    this.pageModel = pageModel;
  }

  @GetMapping("/about")
  public String about(Model model) {
    pageModel.apply(model, new PageMeta("/about", "About · dunamismax", site.defaultDescription(), "about"));
    model.addAttribute("aboutHtml", content.aboutHtml());
    return "pages/about";
  }
}

File: src/main/kotlin/com/dunamismax/site/web/AboutController.kt
package com.dunamismax.site.web

import com.dunamismax.site.config.SiteProperties
import com.dunamismax.site.content.ContentService
import com.dunamismax.site.content.PageMeta
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.GetMapping

@Controller
class AboutController(
    private val content: ContentService,
    private val site: SiteProperties,
    private val pageModel: PageModel,
) {
    @GetMapping("/about")
    fun about(model: Model): String {
        pageModel.apply(
            model,
            PageMeta(
                path = "/about",
                title = "About · dunamismax",
                description = site.defaultDescription,
                section = "about",
            ),
        )
        model.addAttribute("aboutHtml", content.aboutHtml())
        return "pages/about"
    }
}
